File: Backend/src/models/Message.js
const { Schema, model, Types } = require('mongoose');

const MessageSchema = new Schema(
  {
    conversation: { type: Types.ObjectId, ref: 'Conversation', required: true },
    sender: { type: Types.ObjectId, ref: 'User', required: true },
    content: { type: String, required: true },
    readAt: { type: Date }
  },
  { timestamps: true }
);

MessageSchema.index({ conversation: 1, createdAt: -1 });

module.exports = model('Message', MessageSchema);

File: Movil/lib/models/message.dart
import 'package:json_annotation/json_annotation.dart';
import 'user.dart';

part 'message.g.dart';

/// Modelo de Mensaje dentro de conversaciones
@JsonSerializable(explicitToJson: true)
class Message {
  @JsonKey(name: '_id')
  final String id;
  
  @JsonKey(name: 'conversation')
  final String conversationId;
  final User sender;
  final String content;
  final DateTime? readAt;
  final DateTime createdAt;

  Message({
    required this.id,
    required this.conversationId,
    required this.sender,
    required this.content,
    this.readAt,
    required this.createdAt,
  });

  factory Message.fromJson(Map<String, dynamic> json) =>
      _$MessageFromJson(json);
  Map<String, dynamic> toJson() => _$MessageToJson(this);
  
  /// Verificar si el mensaje fue leído
  bool get isRead => readAt != null;
}
